anta/tests/bfd.py

```diff
@@ -0,0 +1,152 @@
+    commands = [AntaCommand(command="show bfd peers")]
+
+    class Input(AntaTest.Input):
+        """
+        This class defines the input parameters of the testcase.
+        """
+
+        bfd_peers: List[BFDPeers]
+        """List of BFD peers"""
+
+        class BFDPeers(BaseModel):
+            """
+            This class defines the details of a BFD peer.
+            """
+
+            peer_address: IPv4Address
+            """IPv4 address of a BFD peer"""
+            vrf: str = "default"
+            """Optional VRF for BGP peer. If not provided, it defaults to `default`."""
+
+    @AntaTest.anta_test
+    def test(self) -> None:
+        failures: dict[Any, Any] = {}
+
+        # Iterating over BFD peers
+        for bfd_peer in self.inputs.bfd_peers:
+            peer = str(bfd_peer.peer_address)
+            vrf = bfd_peer.vrf
+            bfd_output = get_value(self.instance_commands[0].json_output, f"vrfs..{vrf}..ipv4Neighbors..{peer}..peerStats..", separator="..")
+
+            # Check if BFD peer configured
+            if not bfd_output:
+                failures[peer] = {vrf: "Not Configured"}
+                continue
+
+            # Check BFD peer status and remote disc
+            if not (bfd_output.get("status") == "up" and bfd_output.get("remoteDisc") != 0):
+                failures[peer] = {vrf: {"status": bfd_output.get("status"), "remote_disc": bfd_output.get("remoteDisc")}}
+
+        if not failures:
+            self.result.is_success()
+        else:
+            self.result.is_failure(f"Following BFD peers are not configured, status is not up or remote disc is zero:\n{failures}")
+
+
+class VerifyBFDPeersIntervals(AntaTest):
+    """
+    This class verifies the timers of the BFD peers in the specified VRF.
+
+    Expected results:
+        * success: The test will pass if the timers of the BFD peers are correct in the specified VRF.
+        * failure: The test will fail if the BFD peers are not found or their timers are incorrect in the specified VRF.
+    """
+
+    name = "VerifyBFDPeersIntervals"
+    description = "Verifies the timers of the BFD peers in the specified VRF."
+    categories = ["bfd"]
+    commands = [AntaCommand(command="show bfd peers detail")]
+
+    class Input(AntaTest.Input):
+        """
+        This class defines the input parameters of the testcase.
+        """
+
+        bfd_peers: List[BFDPeers]
+        """List of BFD peers"""
+
+        class BFDPeers(BaseModel):
+            """
+            This class defines the details of a BFD peer.
+            """
+
+            peer_address: IPv4Address
+            """IPv4 address of a BFD peer"""
+            vrf: str = "default"
+            """Optional VRF for BGP peer. If not provided, it defaults to `default`."""
+            tx_interval: int
+            """Tx interval of BFD peer"""
+            rx_interval: int
+            """Rx interval of BFD peer"""
+            multiplier: int
+            """Multiplier of BFD peer"""
```

Comment:
Can we be more restrictive for these inputs?

Reply:
Yes we can, added intervals and multiplier in custom output